.sencha/app/cordova-impl.xml

```diff
@@ -0,0 +1,180 @@
+
+            if (available >= 0) {
+                existing_platforms = existing_platforms.substring(0, available);
+            }
+
+            var missing_platforms = [], platformIndex, platform;
+            for (platformIndex in target_platforms) {
+                platform = target_platforms[platformIndex];
+                if(existing_platforms.indexOf(platform) === -1){
+                    missing_platforms.push(platform);
+                }
+            }
+
+            if(missing_platforms.length > 0) {
+                self.log("Cordova: Missing platforms " + missing_platforms.toString() + ". Attempting add.");
+                project.setProperty("cordova.platforms.missing", missing_platforms.join(" "));
+            } else if(existing_platforms.length() >0){
+                self.log("Cordova: All requested platforms exist");
+            } else {
+                self.log("Cordova: No platforms exist");
+            }
+        </script>
+
+        <if>
+            <isset property="cordova.platforms.missing"/>
+            <then>
+                <cordova-echo message="Cordova is adding platforms &quot;${cordova.platforms.missing}&quot;"/>
+                <x-shell reloadprofile="true" dir="${app.cordova.config.path}">
+                    cordova ${cordova.cli.options} platform add ${cordova.platforms.missing}
+                </x-shell>
+            </then>
+        </if>
+    </target>
+    <target name="-after-cordova-platform-add"/>
+    <target name="cordova-platform-add" depends="cordova-init, -before-cordova-platform-add, -cordova-platform-add, -after-cordova-platform-add"/>
+
+    <target name="-before-cordova-init-native-package"/>
+    <target name="-cordova-init-native-package" depends="cordova-create"/>
+    <target name="-after-cordova-init-native-package"/>
+    <target name="cordova-init-native-package" depends="cordova-init, -before-cordova-init-native-package, -cordova-init-native-package, -after-cordova-init-native-package"/>
+
+    <target name="-before-cordova-native-package"/>
+    <target name="-cordova-native-package" depends="cordova-platform-add, cordova-build, cordova-prepare, cordova-emulate, cordova-run"/>
+    <target name="-after-cordova-native-package"/>
+    <target name="cordova-native-package" depends="cordova-init, -before-cordova-native-package, -cordova-native-package, -after-cordova-native-package"/>
+
+
+    <macrodef name="cordova-echo">
+        <text name="text" optional="true"/>
+        <attribute name="message" default=""/>
+        <attribute name="level" default="info"/>
+        <sequential>
+            <if>
+                <equals arg1="@{message}" arg2=""/>
+                <then>
+                    <x-echo message="[Cordova] @{text}" level="@{level}"/>
+                </then>
+                <else>
+                    <x-echo message="[Cordova] @{message}" level="@{level}"/>
+                </else>
+            </if>
+        </sequential>
+    </macrodef>
+</project>
```

Comment:
There's no newline at the end of the file. Since we're poking this, we may as well add one.